src/Payments/Integrations/AbstractPaymentMethodType.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Abstract payment method type class.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\Payments\Integrations;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Blocks\Payments\PaymentMethodTypeInterface;
+
+/**
+ * AbstractPaymentMethodType class.
+ *
+ * @since $VID:$
+ */
+abstract class AbstractPaymentMethodType implements PaymentMethodTypeInterface {
```

Comment:
You may wonder why have both an interface and an AbstractPaymentMethodType?  The primary reason is flexibility for how payment method extensions set things up. We _may_ add helpers or extra things to the AbstractPaymentMethodType, but the interface defines the contract needed for the api. So payment method extensions can just implement the interface if they want to do their own thing.